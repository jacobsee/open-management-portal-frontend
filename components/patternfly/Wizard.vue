<template>
  <section class="pf-c-page__main-wizard">
    <div class="pf-c-wizard pf-m-in-page fill-wizard">
      <button
        aria-label="Wizard Header Toggle"
        class="pf-c-wizard__toggle"
        aria-expanded="false"
      >
        <ol class="pf-c-wizard__toggle-list">
          <li class="pf-c-wizard__toggle-list-item">
            <span class="pf-c-wizard__toggle-num">2</span>Point of Contact
            <!-- <i
              class="fas fa-angle-right pf-c-wizard__toggle-separator"
              aria-hidden="true"
            >
            </i> -->
          </li>
          <!-- <li class="pf-c-wizard__toggle-list-item">Substep B</li> -->
        </ol>
        <i
          class="fas fa-caret-down pf-c-wizard__toggle-icon"
          aria-hidden="true"
        >
        </i>
      </button>
      <div class="pf-c-wizard__outer-wrap">
        <div class="pf-c-wizard__inner-wrap">
          <nav class="pf-c-wizard__nav" aria-label="Steps">
            <ol class="pf-c-wizard__nav-list">
              <WizardNavItem
                v-for="item in navigation"
                :key="item.key"
                :is-current="item.key === $data.currentState"
                :item="item"
                @navigate="navigate"
              >
                {{ item.displayName }}
              </WizardNavItem>
            </ol>
          </nav>
          <div class="pf-c-wizard__main">
            <div class="pf-c-wizard__main-body">
              <slot :name="`${$data.currentState}`" />
            </div>
          </div>
        </div>
        <footer class="pf-c-wizard__footer">
          <button
            class="pf-c-button pf-m-primary"
            type="submit"
            :disabled="currentState === navigation[navigation.length - 1].key"
            @click="next"
          >
            Next
          </button>
          <button
            class="pf-c-button pf-m-secondary"
            type="button"
            :disabled="currentState === navigation[0].key"
            @click="prev"
          >
            Back
          </button>
          <button class="pf-c-button pf-m-link" type="button">Cancel</button>
        </footer>
      </div>
    </div>
  </section>
</template>

<script>
import WizardNavItem from '~/components/patternfly/WizardNavItem.vue'

export default {
  components: {
    WizardNavItem
  },
  props: ['navigation', 'initialState'],
  data() {
    return {
      currentState: this.initialState
    }
  },
  methods: {
    navigate(state) {
      this.currentState = state
    },
    next() {
      for (let i = 0; i < this.navigation.length; i++) {
        const item = this.navigation[i]
        if (item.key === this.currentState && i < this.navigation.length - 1) {
          this.currentState = this.navigation[i + 1].key
          i = this.navigation.length
        }
      }
    },
    prev() {
      for (let i = 0; i < this.navigation.length; i++) {
        const item = this.navigation[i]
        if (item.key === this.currentState && i > 0) {
          this.currentState = this.navigation[i - 1].key
          i = this.navigation.length
        }
      }
    }
  }
}
</script>

<style>
.fill-wizard {
  min-height: calc(100vh - 76px - 112px);
  /*                ^ 100% window height
                            ^ header height
                                    ^ title height
  */
}
</style>
