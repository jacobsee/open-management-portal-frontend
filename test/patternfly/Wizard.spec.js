import { mount } from '@vue/test-utils'
import Wizard from '@/components/patternfly/Wizard.vue'

const testProps = {
  navigation: [
    {
      key: 'first-form',
      displayName: 'First Form'
    },
    {
      key: 'second-form',
      displayName: 'Second Form'
    }
  ]
}

describe('Wizard', () => {
  test('is a Vue instance', () => {
    const wrapper = mount(Wizard, {
      propsData: testProps
    })
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  test('renders properly', () => {
    const wrapper = mount(Wizard, {
      propsData: testProps
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
