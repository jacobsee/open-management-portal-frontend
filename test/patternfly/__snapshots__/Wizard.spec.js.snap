// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wizard renders properly 1`] = `
"<section class=\\"pf-c-page__main-wizard\\"><div class=\\"pf-c-wizard pf-m-in-page fill-wizard\\"><button aria-label=\\"Wizard Header Toggle\\" aria-expanded=\\"false\\" class=\\"pf-c-wizard__toggle\\"><ol class=\\"pf-c-wizard__toggle-list\\"><li class=\\"pf-c-wizard__toggle-list-item\\"><span class=\\"pf-c-wizard__toggle-num\\">2</span>Point of Contact
          </li></ol> <i aria-hidden=\\"true\\" class=\\"fas fa-caret-down pf-c-wizard__toggle-icon\\"></i></button> <div class=\\"pf-c-wizard__outer-wrap\\"><div class=\\"pf-c-wizard__inner-wrap\\"><nav aria-label=\\"Steps\\" class=\\"pf-c-wizard__nav\\"><ol class=\\"pf-c-wizard__nav-list\\"><li class=\\"pf-c-wizard__nav-item\\"><a class=\\"pf-c-wizard__nav-link\\">
              First Form
            </a></li><li class=\\"pf-c-wizard__nav-item\\"><a class=\\"pf-c-wizard__nav-link\\">
              Second Form
            </a></li></ol></nav> <div class=\\"pf-c-wizard__main\\"><div class=\\"pf-c-wizard__main-body\\"></div></div></div> <footer class=\\"pf-c-wizard__footer\\"><button type=\\"submit\\" class=\\"pf-c-button pf-m-primary\\">
          Next
        </button> <button type=\\"button\\" class=\\"pf-c-button pf-m-secondary\\">
          Back
        </button> <button type=\\"button\\" class=\\"pf-c-button pf-m-link\\">Cancel</button></footer></div></div></section>"
`;
